data/data.js

```diff
@@ -1,7 +1,74 @@
+const fs =  require('fs');
+const auth = require('../utils/auth');
+
+
+
+function addItem(req, res, obj, file) {
+    auth.authenticate(req, res, function () {
+        var item = req.body;
+        if (item.id === '-1') {
+            item.id = obj.length.toString();
+            obj.push(item);
+        } else {
+            let idx = findIdx(obj, item.id);
+            if (idx > -1) {
+                obj[idx] = item;
+            } else {
+                return res.status(404).json({ error: 'ID not found' });
+            }
+        }
+        fs.writeFile(file, JSON.stringify(obj, null, 4), function (error) {
```

Comment:
Might be worth separating this into a standalone function as the way both `addItem` and `deleteItem` write to file the same way.

Reply:
yup code duplicate said so